### PR TITLE
Fix reported properties sync bug

### PR DESCRIPTION
**Pull request by Mandur**
Our current implementation of the additional join properties was not syncing the reported properties across different gateway. Hence it was non working in multi gateway scenarios.

### CODE REVIEW

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs [hunk 1]
```diff
@@ -560,9 +576,9 @@ public async Task<bool> SaveChangesAsync(TwinCollection reportedProperties = nul
                 var fcntDownDelta = this.FCntDown >= this.LastSavedFCntDown ? this.FCntDown - this.LastSavedFCntDown : this.LastSavedFCntDown - this.FCntDown;
 
                 if (reportedProperties.Count > 0 ||
-                    fcntDownDelta >= Constants.MAX_FCNT_UNSAVED_DELTA ||
-                    fcntUpDelta >= Constants.MAX_FCNT_UNSAVED_DELTA ||
-                    (this.hasFrameCountChanges && force))
+                            fcntDownDelta >= Constants.MAX_FCNT_UNSAVED_DELTA ||
+                            fcntUpDelta >= Constants.MAX_FCNT_UNSAVED_DELTA ||
+                            (this.hasFrameCountChanges && force))
```
Comment (debugerr) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaDevice.cs]: I prefer the othe rstyle where things are aligned with the first line

**Comment by TechPreacher** [comment]
Looks good!

**Comment by fbeltrao** [comment]
My feedback is to provide a description of what this PR is supposed to do. Is the problem that we handle RX delay properties in all devices types, instead of OTAA only?

**Comment by ronniesa** [comment]
> My feedback is to provide a description of what this PR is supposed to do. Is the problem that we handle RX delay properties in all devices types, instead of OTAA only?

No there was a bug that on first message we did not read the reported twin for the RX delay. So the cache version after a join works but in case of a ns restart or a multigateway it would fail.

**Comment by Mandur** [comment]
I forced push this branch to clean up history. It is now ready for test and review.

**Pull request merged**